pkg/webhook/storageclass.go

```diff
@@ -27,152 +29,96 @@ import (
-	reviewResponse := &v1.AdmissionResponse{
-		Allowed: true,
-		Result:  &metav1.Status{},
-	}
-
-	if isUpdate {
-		// TBD
-		return reviewResponse
-	}
-
-	if err := ValidateV1StorageClass(sc); err != nil {
-		reviewResponse.Allowed = false
-		reviewResponse.Result.Message = err.Error()
-	}
-	return reviewResponse
-}
-
-func ValidateV1StorageClass(sc *storagev1.StorageClass) error {
-	if sc == nil {
-		return fmt.Errorf("StorageClass is nil")
-	}
-
-	if sc.Provisioner != FilestoreCSIDriver {
-		return nil
-	}
-
-	params := sc.Parameters
-	if params == nil {
-		return nil
-	}
-
-	if _, ok := params["multishare"]; ok {
-		klog.Infof("Multishare Filestore CSI storage class detected")
-	}
-
-	// TBD: Fill the validation logic here.
-	return nil
-}
-
-func toV1AdmissionResponse(err error) *v1.AdmissionResponse {
+func rejectV1AdmissionResponse(err error) *v1.AdmissionResponse {
 	return &v1.AdmissionResponse{
 		Result: &metav1.Status{
 			Message: err.Error(),
 		},
 	}
 }
+
 func mutateStorageClass(ar v1.AdmissionReview) *v1.AdmissionResponse {
 	klog.Info("mutating storageClass")
 	reviewResponse := &v1.AdmissionResponse{
 		Allowed: true,
 		Result:  &metav1.Status{},
 	}
 
-	if !(ar.Request.Operation == v1.Update || ar.Request.Operation == v1.Create) {
+	if ar.Request.Operation != v1.Create {
 		return reviewResponse
 	}
 
 	raw := ar.Request.Object.Raw
-	oldRaw := ar.Request.OldObject.Raw
 
 	deserializer := codecs.UniversalDeserializer()
 	switch ar.Request.Resource {
 	case StorageClassV1GVR:
 		sc := &storagev1.StorageClass{}
 		if _, _, err := deserializer.Decode(raw, nil, sc); err != nil {
 			klog.Error(err)
-			return toV1AdmissionResponse(err)
-		}
-		oldSc := &storagev1.StorageClass{}
-		if _, _, err := deserializer.Decode(oldRaw, nil, oldSc); err != nil {
-			klog.Error(err)
-			return toV1AdmissionResponse(err)
+			return rejectV1AdmissionResponse(err)
 		}
 		klog.Infof("check patch for storageClass %s", sc.Name)
 		return applyV1StorageClassPatch(sc)
 	default:
-		err := fmt.Errorf("expect resource to be %s", StorageClassV1GVR)
+		err := fmt.Errorf("expect resource to be %v", StorageClassV1GVR)
 		klog.Error(err)
-		return toV1AdmissionResponse(err)
+		return rejectV1AdmissionResponse(err)
 	}
 }
 
 func applyV1StorageClassPatch(sc *storagev1.StorageClass) *v1.AdmissionResponse {
 	reviewResponse := &v1.AdmissionResponse{
 		Allowed: true,
+		Result:  &metav1.Status{},
 	}
 
 	if sc.Provisioner != FilestoreCSIDriver {
 		return reviewResponse
 	}
 
-	isMultishare, ok := sc.Parameters["multishare"]
-	if !ok || isMultishare == "false" || isMultishare == "False" {
+	isMultishare, ok := sc.Parameters[Multishare]
+	if !ok || strings.ToLower(isMultishare) == "false" {
 		return reviewResponse
 	}
 
-	if _, ok := sc.Parameters["instancePrefix"]; ok {
-		return reviewResponse
+	if strings.ToLower(isMultishare) != "true" {
+		return rejectV1AdmissionResponse(fmt.Errorf("the value for %q can only be True or False", Multishare))
```

Comment:
"acceptable values for % is true|True|false|False"

Reply:
done